FileAPI/support/Blob.js

```diff
@@ -5,49 +5,35 @@ self.test_blob = (fn, expectations) => {
       type = expectations.type,
       desc = expectations.desc;
 
-  var t = async_test(desc);
-  t.step(function() {
+  promise_test(async (t) => {
     var blob = fn();
     assert_true(blob instanceof Blob);
     assert_false(blob instanceof File);
     assert_equals(blob.type, type);
     assert_equals(blob.size, expected.length);
 
-    var fr = new FileReader();
-    fr.onload = t.step_func_done(function(event) {
-      assert_equals(this.result, expected);
-    }, fr);
-    fr.onerror = t.step_func(function(e) {
-      assert_unreached("got error event on FileReader");
-    });
-    fr.readAsText(blob, "UTF-8");
-  });
+    const text = await blob.text();
+    assert_equals(text, expected);
+  }, desc);
 }
 
 self.test_blob_binary = (fn, expectations) => {
   var expected = expectations.expected,
       type = expectations.type,
       desc = expectations.desc;
 
-  var t = async_test(desc);
-  t.step(function() {
+  promise_test(async (t) => {
     var blob = fn();
     assert_true(blob instanceof Blob);
     assert_false(blob instanceof File);
     assert_equals(blob.type, type);
     assert_equals(blob.size, expected.length);
 
-    var fr = new FileReader();
-    fr.onload = t.step_func_done(function(event) {
-      assert_true(this.result instanceof ArrayBuffer,
-                  "Result should be an ArrayBuffer");
-      assert_array_equals(new Uint8Array(this.result), expected);
```

Comment:
I couldn't find another place that asserts the bytes given by `readAsArrayBuffer()` are as expected, like this does. FileAPI/reading-data-section/filereader_result.any.js is the logical place, and it validates the contents of `readAsText()`, `readAsDataURL()`, `readAsBinaryString()`, but not `readAsArrayBuffer()` - just that it is an `ArrayBuffer`.

Can you add an equivalent `assert_array_equals` over there?